src/arch/amdgpu/vega/tlb_coalescer.cc

```diff
@@ -236,6 +266,76 @@ VegaTLBCoalescer::updatePhysAddresses(PacketPtr pkt)
         schedule(cleanupEvent, curTick());
 }
 
+// re-coalesce packet to 4k pages
+void
+VegaTLBCoalescer::reissue_pkt_helper(PacketPtr pkt)
+{
+    // first packet of a coalesced request
+    PacketPtr first_packet = nullptr;
+    // true if we are able to do coalescing
+    bool didCoalesce = false;
+    // number of coalesced reqs for a given window
+    int coalescedReq_cnt = 0;
+
+    GpuTranslationState *sender_state =
+        safe_cast<GpuTranslationState*>(pkt->senderState);
+
+    DPRINTF(GPUTLB, "Trying to re-issue req at tick: %llu, addr: %#x\n",
+            sender_state->issueTime, pkt->req->getVaddr());
+
+    // The tick index is used as a key to the coalescerFIFO hashmap.
+    // It is shared by all candidates that fall within the
+    // given coalescingWindow.
+    Tick tick_index = sender_state->issueTime / coalescingWindow;
+
+    if (coalescerFIFO.count(tick_index)) {
+        coalescedReq_cnt = coalescerFIFO[tick_index].size();
+    }
+
+    // see if we can coalesce the incoming pkt with another
+    // coalesced request with the same tick_index
```

Comment:
Why does tick index matter for coalescing?  Wouldn't any tick within a window be ok?